pkg/labels/labels.go

```diff
@@ -252,9 +252,21 @@ func (l *Label) IsReservedSource() bool {
 	return l.Source == LabelSourceReserved
 }
 
-// matches returns true if l matches the target
-func (l *Label) matches(target *Label) bool {
-	return l.Equals(target)
+// Matches returns true if l Matches the target.
+// note that this is a loose match:
+// - l.Source == any means sources must not match
+func (l *Label) Matches(target *Label) bool {
+	return l.MatchesKey(target) && l.Value == target.Value
+}
+
+// MatchesKey returns true if l has target's key.
+// note that l may be looser than target.
+func (l *Label) MatchesKey(target *Label) bool {
+	if !l.IsAnySource() && l.Source != target.Source {
```

Comment:
> Label matching is directional / non-communtative. Specifically,
`"any:foo=bar".Equals("k8s:foo=bar")` is true, whereas
`"k8s:foo=bar".Equals("any:foo=bar")` is false.

What about `"any:foo=bar".Equals("any:foo=bar")` ? That probably never happens though :thinking: .